infra/lib/stack.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as cdk from "aws-cdk-lib";
+import { CfnOutput } from "aws-cdk-lib";
+import * as wafv2 from "aws-cdk-lib/aws-wafv2";
+import { Construct } from "constructs";
+import { AnalyticsIntegrationTestStack } from "./analytics/stack";
+import {
+  AuthIntegrationTestStack,
+  AuthIntegrationTestStackEnvironmentProps
+} from "./auth/stack";
+import { AmplifyCategory, IntegrationTestStack } from "./common";
+import {
+  StorageAccessLevel,
+  StorageIntegrationTestStack
+} from "./storage/stack";
+
+export class AmplifyInfraStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    // Allow overriding which categories to deploy
+    const categoryParam = this.node.tryGetContext("category");
+    const categories: AmplifyCategory[] = categoryParam
+      ? [categoryParam as AmplifyCategory]
+      : Object.values(AmplifyCategory);
+
+    console.log(`Deploying stacks for categories: ${categories}`);
+
+    // Create shared infrastructure
+
+    // Create a Web Application Firewall (WAF) for blocking malicious requests.
+    const waf = new wafv2.CfnWebACL(this, "WAF", {
+      scope: "REGIONAL",
+      defaultAction: {
+        allow: {},
+      },
+      rules: [
+        // Basic rate limiting to prevent overuse of endpoints
+        {
+          name: "RateLimit",
+          priority: 30,
+          action: {
+            block: {},
+          },
+          statement: {
+            rateBasedStatement: {
+              aggregateKeyType: "IP",
+              // The number of requests which can be performed by
+              // a single IP in a 5-minute window.
+              limit: 1000,
+            },
+          },
+          visibilityConfig: {
+            sampledRequestsEnabled: true,
+            cloudWatchMetricsEnabled: true,
+            metricName: "RateLimit",
+          },
+        },
+      ],
+      visibilityConfig: {
+        sampledRequestsEnabled: false,
+        cloudWatchMetricsEnabled: true,
+        metricName: "WAFViolations",
+      },
+    });
+
+    // The Analytics stack
+    if (categories.includes(AmplifyCategory.Analytics)) {
```

Comment:
question: Are these individual category stacks different from previous ones other than being moved and using WAF?

Reply:
Nope, the only thing updated in the categories was the naming and organization. No new resources.